dallinger/recruiters.py

```diff
@@ -727,6 +736,60 @@ def get_qualifications(self, hit_id, sandbox):
             "peripheral_requirements": details["peripheral_requirements"],
         }
 
+    def screen_out(self, participant: Participant):
+        """Screen-out a participant.
+
+        Parameters:
+            participant: Participant
+                The participant to screen-out.
+        """
+        if self.screen_out_allowed(participant):
+            try:
+                return self.prolificservice.screen_out(
+                    study_id=self.current_study_id,
+                    submission_id=participant.assignment_id,
+                    bonus_per_submission=participant.base_pay + participant.bonus,
+                    increase_places=self.config.get("auto_recruit"),
+                )
+            except ProlificServiceException as ex:
+                logger.exception(str(ex))
+
+    def screen_out_allowed(self, participant):
+        """Check if a participant can be screened out.
+
+        Returns False if:
+        - participant is not in 'working' status
+        - custom screening is enabled
+        - reward is below minimum required
+        """
+        if participant.status != "working":
+            return False
+
+        study = self.prolificservice.get_study(self.current_study_id)
+
+        if study.get("is_custom_screening", False):
+            raise ProlificRecruiterException(
+                f"Prolific study (ID {self.current_study_id}) doesn't allow screening-out of participants"
+            )
+
+        # Minimum wage thresholds: https://researcher-help.prolific.com/en/article/2273bd
+        prolific_min_wage_per_hour = 6
+        participants = Participant.query.filter_by(status="working").all()
```

Comment:
I believe that the screen_out route should only look at the participants you have selected within that specific screen_out call, not at the full set of participants

Reply:
These participants are only retrieved from the database to calculate the `median_time_spent_in_hours` based on the description here:

> Prolific has some internal logic that decides whether to approve the `screen_out` request. From what I understand, it examines the list of participants, calculates their median time in the experiment, and then compares this to the amount of money which you are proposing to pay them. If the money is lower than the time \* the minimum wage per hour that Prolific permits, then it rejects the call. We should implement this logic in a helper function called `screen_out_allowed`.

See https://gitlab.com/PsyNetDev/PsyNet/-/issues/881

Reply:
Ah, I see now.

Reply:
But then, we normally would try to only screen out one participant at a time, don't we?

Reply:
Sure -- so in practice, it will often just be checking that single participant's time. But in theory we could use this recruiter method in different ways in the future

Reply:
Done.